sklearn/linear_model/_sgd_fast.pyx.tp

```diff
@@ -475,7 +479,23 @@ def _plain_sgd{{name_suffix}}(
                     eta = eta0 / pow(t, power_t)
 
                 if verbose or not early_stopping:
-                    sumloss += loss.cy_loss(y, p)
+                    cur_loss_val = loss.cy_loss(y, p)
+                    sumloss += cur_loss_val
+                    if learning_rate != PA1 and learning_rate != PA2:
+                        # sum up the optimization objective function terms (not just loss, include regularization)
+                        # SGD optimizes 0.5 * L2**2, because using weight decay
+                        objective_sum += cur_loss_val
+                        if penalty_type > 0: # if regularization is enabled
+                            objective_sum + alpha * (
```

Comment:
```suggestion
                            objective_sum += alpha * (
```

??? And I think we should make sure to have a test which would fail before this fix